docs/guides/transactions/running-background-jobs.md

```diff
@@ -2,41 +2,34 @@
  
 ## Introduction
 
-Code that is provided via Starcounter API (transactions, SQL, request/response handling, Web sockets handling, etc.) should execute in a special context. Such context is already given inside the `Main()` function, view-model handles and handlers (for example, `Handle.GET`). In other situations, like in separately created background .NET threads or tasks, to be able to execute Starcounter code, one should schedule it using `Scheduling.RunTask()`.
+Starcounter maintains its own thread pool. Most of Starcounter API may be called only from a thread from the Starcounter thread pool. Among such APIs are: methods from Starcounter.Db class, methods from Starcounter.Request class etc. Safer to regard all APIs from Starcounter namespace as required to be called on a Starcounter thread, unless explicitly stated otherwise. Starcounter assures that an application `Main`, view-model handles and HTTP-handlers are called on a Starcounter thread. In other situations, like your code is running on separately created .Net threads or on .Net thread pool, you have to schedule a task on Starcounter thread pool using `Scheduling.RunTask` in order to be able to use Starcounter API. `Scheduling.RunTask` is one of exceptional API's that can be called on any thread. Another notable exception is `StarcounterEnvironment.IsOnScheduler()` which is to determine whether you're on a Starcounter thread.
```

Comment:
This seems to cover it well. I just have a couple of questions:

* What's "view-model handles"?
* Are "HTTP-handlers" just HTTP handlers registered with the `Handle` API?
* With "is one of exceptional API's", do you mean that it's an exception to the rule that Starcounter APIs should be called with a Starcounter thread? 

If you rephrase these things slightly so that they're easier to understand, I think this will be a good introduction paragraph.

There are still some grammar and style mistakes, I'll fix them in a later PR. Pointing out every single one and then fixing them would be too time consuming.

Reply:
> What's "view-model handles"?

Just a copy-paste from the previous version. I thought we use this term to refer to typed json event handlers. If not, we might want to change it. cc @alemoi 

>Are "HTTP-handlers" just HTTP handlers registered with the Handle API?

I meant this, as we currently support HTTP handlers only (AFAIK). Up to @alemoi 

>With "is one of exceptional API's", do you mean that it's an exception to the rule that Starcounter APIs should be called with a Starcounter thread?

Exaclty. Please advise on wording.

Reply:
> Just a copy-paste from the previous version. I thought we use this term to refer to typed json event handlers. If not, we might want to change it

Fair enough, then I would write "Typed JSON event handlers". That's the term we use in the tutorial.

> I meant this, as we currently support HTTP handlers only (AFAIK).

That makes sense. Then, I would write "HTTP-handlers registered with the `Handle` API", just to make it clear.

> Exaclty. Please advise on wording.

What about something like this?

> You can call `Scheduling.RunTask` while on a .NET or Starcounter thread. `StarcounterEnvironment.IsOnScheduler` is another method that can be called on any thread, it tells you if the current thread is a Starcounter thread.

Reply:
All requested changes applied. Please approve and we merge into my branch.